build/Linux+BSD/portable/x86_64/Recipe

```diff
@@ -1,92 +1,100 @@
+  curl
+  libasound2-dev # ALSA
+  libfontconfig1-dev
+  libfreetype6-dev
+  libgl1-mesa-dev
+  libjack-dev
+  libmp3lame-dev
+  libnss3-dev
+  libportmidi-dev
+  libpulse-dev
+  libsndfile1-dev
+  make
+  portaudio19-dev
+  wget
+  )
+
+# MuseScore compiles without these but won't run without them
+apt_packages_runtime=(
+  # alphabetical order please
+  libegl1-mesa-dev
+  libodbc1
+  libpq-dev
+  libxcomposite-dev
+  libxcursor-dev
+  libxi-dev
+  libxkbcommon-x11-0
+  libxtst-dev
+  )
 
 apt-get update # no package lists in Docker image
+apt-get install -y "${apt_packages_basic[@]}" "${apt_packages_standard[@]}" "${apt_packages_runtime[@]}"
 
-# basic dependencies (needed by Docker but provided on Travis)
-apt-get -y install \
-  git \
-  unzip \
-  libxslt1-dev \
-  libxrender-dev \
-  libxcomposite-dev \
-  libdrm-dev \
-  libgl1-mesa-dev \
-  libopenal1 \
-  gstreamer0.10-plugins-base \
-  libmirclient9 \
-  software-properties-common # installs `add-apt-repository`
+# NON-DISTRIBUTION PACKAGES
+# We use these when the default version provided by the distribution is
+# too old or has unwelcome modifications compared to the upstream version.
 
 # get newer compiler
-if [ ! -f /etc/apt/sources.list.d/ubuntu-toolchain-r-test-precise.list ]; then
+if [[ ! -f /etc/apt/sources.list.d/ubuntu-toolchain-r-test-*.list ]]; then
   add-apt-repository -y ppa:ubuntu-toolchain-r/test
   apt-get update
 fi
+apt-get install -y g++-7
+update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 40 --slave /usr/bin/g++ g++ /usr/bin/g++-7
 
-# MuseScore dependencies (same as on Travis)
-apt-get -y install \
-  g++-4.9 \
-  wget \
-  make \
-  cmake \
-  curl \
-  alsa \
-  libasound2-dev \
-  portaudio19-dev \
-  libportmidi-dev \
-  libsndfile1-dev \
-  zlib1g-dev \
-  libfreetype6-dev \
-  libfontconfig1-dev \
-  lame \
-  libmp3lame-dev \
-  libegl1-mesa-dev \
-  libpulse-dev \
-  libnss3-dev \
-  libxss1 \
-  libnspr4-dev \
-  libxi-dev \
-  libxcursor-dev \
-  libxtst-dev
-
-
-update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.9 40 --slave /usr/bin/g++ g++ /usr/bin/g++-4.9
-
-# get Qt
-if [ ! -d "qt5" ]; then
+# get newer Qt
+if [[ ! -d "qt5" ]]; then
   mkdir qt5
   wget -q -O qt5.zip https://s3.amazonaws.com/utils.musescore.org/qt5124.zip
   unzip -qq qt5.zip -d qt5
   rm -f qt5.zip
 fi
-export PATH="${PWD}/qt5/bin:$PATH"
-export QT_PLUGIN_PATH="${PWD}/qt5/plugins"
-export QML2_IMPORT_PATH="${PWD}/qt5/qml"
-export LD_LIBRARY_PATH="${PWD}/qt5/lib:$LD_LIBRARY_PATH"
-
-# build AppImageKit
-if [ ! -d "AppImageKit" ]; then
-  git clone https://github.com/probonopd/AppImageKit.git
-  cd AppImageKit
-  git reset --hard 076a88a57d7ca1a8d77bdbf7d8e92f37eea03dab
-  rm -rf .git*
-  ./build.sh
-  cd ..
+qt_dir="${PWD}/qt5"
+export PATH="${qt_dir}/bin:${PATH}"
+export LD_LIBRARY_PATH="${qt_dir}/lib:${LD_LIBRARY_PATH}"
+export QT_PLUGIN_PATH="${qt_dir}/plugins"
+export QML2_IMPORT_PATH="${qt_dir}/qml"
+
+# get newer CMake
+if [[ ! -d cmake ]]; then
```

Comment:
Why don't you want to use CMake fetched with the apt-get?

Reply:
I was following @lasconic's example. He doesn't like distribution packages because they tend to be out of date, and sometimes the are modified and behave differently compared to the upstream version.

It's probably not as important for CMake as for Qt though, so it's up to you which you prefer.

Reply:
For reference, the versions available on Ubuntu 16.04 are:

- CMake: 3.5 (upstream is 3.14)
- GCC: 4.5 (upstream is 9.1)
- Qt: 5.5 (upstream is 5.14)

So fairly ancient in each case.

The consequences for users are:

- __CMake:__
    - This is purely a build tool (doesn't affect users).
    - *Bottom line: We don't care which version is used as long as it builds.*
- __GCC:__
    - Using a newer compiler means users might not have the right libraries pre-installed.
    - It will still work, it just means bundling more libraries in the AppImage (e.g. `libstdc++`).
    - *Bottom line: We must balance features with filesize.*
- __Qt:__
    - Most distributions do not come with Qt installed by default.
    - This means we have no choice but to bundle it in the AppImage.
    - *Bottom line: We might as well bundle the most recent version.*

Of course there might be other reasons for choosing a particular version, such as ease of installation for developers, or a desire to use the same version across different platforms, etc.